src/test/parsers/literal.test.ts

```diff
@@ -28,10 +29,30 @@ describe("literalArgumentParser", () => {
                     defined(literalArgumentParser.parse(reader, properties)),
                     true,
                     [],
-                    []
+                    [],
+                    1
                 );
                 assert.strictEqual(reader.cursor, 4);
             });
+            it("should return the correct highlight data", () => {
+                const reader = new StringReader("test");
+                const out = literalArgumentParser.parse(reader, properties);
+                assert.deepStrictEqual(
```

Comment:
Maybe we need a better way of managing testing actions: actions should be tested inside each test, not as a seperate test. Maybe an `AssertActions` type which takes over numActions in assertReturn. Additionally assertReturn's signature should use the object pattern, because there are two many arguments. If you don't mind, I could do that in this branch.